pkg/lightning/mydump/csv_parser.go

```diff
@@ -297,10 +372,13 @@ outside:
 			parser.fieldIndexes = append(parser.fieldIndexes, len(parser.recordBuffer))
 			break outside
 		default:
-			if err = processDefault(firstByte); err != nil {
-				return nil, err
+			if prevToken == csvTokenDelimiter {
+				parser.logSyntaxError()
+				return nil, errors.AddStack(errUnexpectedQuoteField)
 			}
+			parser.appendCSVTokenToRecordBuffer(firstToken)
```

Comment:
If we enter here, we only allow to read a valid unquote field, but this logic will allow read csv with broken syntax. e.g.
```
delimiter = "|+|"
sep = ","
input = "abc|1|+||+|\r\n"

this input should raise an error because delimiter appear in the middle of an unquoted string
```